### ISSUE TITLE
Deserializing with JSON.NET not working for class references

### ISSUE THREAD

**Issue by carldebilly** [nventive/Uno.CodeGen#14]
This won't deserialize :

``` csharp
public class A
{
    public string Str1 { get; }
}

public class B
{
    public A MyA { get; }
}

[...]
var b = JsonConvert.DeserializeObject<B>(json);
```

**Comment by carldebilly** [comment]
Proposed solution would be to generate a [custom converter](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_Converters_CustomCreationConverter_1.htm) when a dependency to JSON.NET is detected.

**Comment by carldebilly** [comment]
Fixed in PR #15
